test/gtest/ucs/test_type.cc

```diff
@@ -138,6 +142,145 @@ UCS_TEST_F(test_type, pack_float) {
               UCS_FP8_PACK_UNPACK(TEST_LATENCY, 200000000));
 }
 
+class test_rwlock : public ucs::test {
+protected:
+    void sleep()
+    {
+        std::this_thread::sleep_for(std::chrono::milliseconds(1));
+    }
+
+    void measure_one(int num, int writers, const std::function<void()> &r,
+                     const std::function<void()> &w, const std::string &name)
+    {
+        std::vector<std::thread> tt;
+
+        tt.reserve(num);
+        auto start = std::chrono::high_resolution_clock::now();
+        for (int c = 0; c < num; c++) {
+            tt.emplace_back([&]() {
+                unsigned seed = time(0);
+                for (int i = 0; i < 1000000 / num; i++) {
+                    if ((rand_r(&seed) % 256) < writers) {
+                        w();
+                    } else {
+                        r();
+                    }
+                }
+            });
+        }
+
+
+        for (auto &t : tt) {
+            t.join();
+        }
+        auto end = std::chrono::high_resolution_clock::now();
+        std::chrono::duration<double> elapsed = end - start;
+
+        UCS_TEST_MESSAGE << elapsed.count() * 1000 << " ms " << name << " "
+                         << std::to_string(num) << " threads "
+                         << std::to_string(writers) << " writers per 256 ";
+    }
+
+    void measure(const std::function<void()> &r,
+                 const std::function<void()> &w, const std::string &name)
+    {
+        int m = std::thread::hardware_concurrency();
+        std::vector<int> threads = {1, 2, 4, m};
+        std::vector<int> writers_per_256 = {1, 25, 128, 250};
+
```

Comment:
1. i think we also want to measure overhead of read lock+unlock regardless of concurrency since it is the reason we added lightweight rwlock
2. can you pls post example output in the PR description?

Reply:
1. you mean write percent 0?
2. didn't i do that?